### PR TITLE
Graded monads

### PR DESCRIPTION
**Pull request by sstucki**
Apparently, I wrote this a year ago but forgot to submit a PR. It's been sitting on a stale branch in my fork of the repo for a year so it needs to be properly rebased before it's ready to be merged. In particular, the first commit (6e5e05e2934da0bff465906dadc4a6d88dc1a5dc) will become redundant once #389 has been merged.

### CODE REVIEW

**Comment by JacquesCarette** [comment]
Thanks for `GradedMonad`, I'm eager to merge that in. I'm not keen on using 'prime' for the name of the alternate definition. Can we try to find a better one? Would `KlesliGradedMonad` be appropriate?

**Comment by JacquesCarette** [comment]
I prefer `GradedKleisliTriple`, it is more consistent.

**Comment by sstucki** [comment]
I've rebased the old commits on the latest master and fixed the name of the alternate definition to `GradedKleisliTriple` as agreed, so this is basically ready to be merged now. I'm just waiting on #389 (not sure what the status is there).

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
One tiny improvement, and this is ready to go - thanks.

File: src/Categories/Category/Monoidal/Construction/Endofunctors.agda
Comment (sstucki) [reply src/Categories/Category/Monoidal/Construction/Endofunctors.agda]: The proof can be shortened down to two lines indeed. I pushed an update with some other optimizations, inspired by the code in #389.

**Comment by sstucki** [comment]
@JacquesCarette, I fixed `Endofuctors.agda` as you suggested, although I thought that file would be replaced by #389, once it was ready. I'm not sure what the status of #389 is, the authors haven't replied to the requests for changes, so I'll leave it up to you if you want to wait for that PR to be fixed, or merge the version from this PR instead (that would make #389 obsolete).

**Comment by JacquesCarette** [comment]
This has waited long enough - I'll merge this in as it is ready.

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**
